### PR TITLE
fix: ethers transactions

### PR DESCRIPTION
**Pull request by tomiir**
# Breaking Changes

N/A

# Changes

- feat:
- fix: add getTransactionByHash to zod validations
- chore:

### CODE REVIEW

File: packages/ethers/package.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
     }
   },
   "optionalDependencies": {
-    "@web3modal/siwe": "3.5.5",
+    "@web3modal/siwe": "3.6.0-2aeae674",
```
Comment (0xAsimetriq) [inline packages/ethers/package.json:81]: Version here (comit hash) looks different from others, an error?

File: packages/ethers5/package.json [hunk 2]
```diff
@@ -78,7 +78,7 @@
     }
   },
   "optionalDependencies": {
-    "@web3modal/siwe": "3.5.5",
+    "@web3modal/siwe": "3.6.0-2aeae674",
```
Comment (0xAsimetriq) [inline packages/ethers5/package.json:81]: Looks like all `@web3modal/siwe` packages have wrong version

File: packages/wallet/src/W3mFrameSchema.ts [hunk 3]
```diff
@@ -146,7 +152,13 @@ export const W3mFrameSchema = {
 
     .or(z.object({ type: zType('FRAME_RPC_REQUEST_ERROR'), payload: zError }))
 
-    .or(z.object({ type: zType('FRAME_RPC_REQUEST_SUCCESS'), payload: RpcResponse }))
+    .or(
+      z.object({
+        type: zType('FRAME_RPC_REQUEST_SUCCESS'),
+        payload: RpcResponse,
+        keepAlive: boolean().optional().default(false)
```
Comment (0xAsimetriq) [inline packages/wallet/src/W3mFrameSchema.ts:159]: Was `z.boolean` not available?

File: packages/ethers/src/client.ts [hunk 4]
```diff
@@ -746,8 +746,18 @@ export class Web3Modal extends Web3ModalScaffold {
 
   private watchEmail() {
     if (this.emailProvider) {
-      this.emailProvider.onRpcRequest(() => {
-        super.open({ view: 'ApproveTransaction' })
+      this.emailProvider.onRpcRequest(request => {
+        const req = request as W3mFrameTypes.AppEvent & { payload?: unknown }
+        const payload = req.payload as W3mFrameTypes.RPCRequest
+        // We only open the modal if it's not a safe (auto-approve) or ignored method
+        const ALLOWED_METHODS = ['eth_blockNumber', 'eth_estimateGas', 'eth_getTransactionByHash']
```
Comment (svenvoskamp) [inline packages/ethers/src/client.ts:753]: I think we should store these values in `W3mFrameConstants.ts`, so secure site and w3m are always in sync. Also better if we adjust these values in the future.

**Review by svenvoskamp — APPROVED** [review]

**Review by svenvoskamp — APPROVED** [review]

**Pull request merged**